File: server/verifypw.py
#!/usr/bin/python3

import dbmanager 
import sys  

def main():
    if len(sys.argv) < 3:
        print('usage: python3 verifypw.py <username> <password>')
        sys.exit(1)

    username, password = sys.argv[1:]
    user = dbmanager.select_user(username)
    _, hashed_password = dbmanager.hash(password, user.salt)
    if user.hashed_password == hashed_password:
        print('verifypw: correct password')
        sys.exit(0)
    else:
        print('verifypw: incorrect password')
        sys.exit(1)

main()

File: src/passwords/verify-pw.cpp
#include "db-manager.h"

int main(int argc, char* argv[]) {
    if (argc < 3) {
        cout << "usage: ./verify-pw <username> <password>" << endl;
        exit(1);
    }

    string username = argv[1];
    string password = argv[2];
    
    if (verify_user(username, password)) {
        cout << "verify-pw: correct password" << endl;
        return 0;
    } else {
        cout << "verify-pw: incorrect username or password" << endl;
        return 1;
    }
}
